### PR TITLE
#96 Frontend tooling

### PR DESCRIPTION
**Pull request by pirog**
@andrewmallis @robLoach @gremy @reynoldsalec @madeofpeople 

Here is a good starting point to standardize our front end toolkit in Kalastack. This is a short term thing for immediate needs. A better solution will be created for KB2 using docker and other deeper magics. 

I need people to QA this pull request to make sure it is consistent and integrates with our current workflows. In practical terms we just want to make sure the versions of all the things here are appropriate. 
##### Checkout this branch

``` bash
$ git checkout -b pirog-frontend 3.x
$ git pull https://github.com/pirog/kalastack.git frontend
```
##### Read the things

 https://github.com/kalamuna/kalastack/wiki/Adding-front-end-tooling

### CODE REVIEW

**Comment by RobLoach** [comment]
Merge branch 'RobLoach-frontend-rikersit' into frontend.

Riker

**Comment by pirog** [comment]
Let's get some thumbs up/thumbs down/comments on this so we can bring Riker into the fold
@andrewmallis @robLoach @gremy @reynoldsalec @madeofpeople

**Comment by thiagodemellobueno** [comment]
So I would urge us to stop using compass (and its dependency on slow slow ruby ) for our sass needs.
We should use node-sass (which is lib-sass a non-ruby C++ implementation).... it's really the difference between 2 second compiles for sass vs 8-14 seconds (using gbz as a benchmark).

It's caught up in the last few months, I've been using it on a few projects since June/July and I'd is production ready hasnt stumbled on any code. It's the future of sass.

Still, we should focus on keeping node and npm up to date, and the rest can be managed per project using npm, packager, gems, etc... thus not tying us to any specific well-crafted-bespoke implementation while 'giving the people what they want'

**Comment by pirog** [comment]
@madeofpeople that sounds reasonable and more in line with our desires to do more node-based front end things. 

@RobLoach do you foresee any significant downsides to @madeofpeople's suggestion? 

I would like this PR to reach https://www.youtube.com/watch?v=vgQidVzdneE status fairly soon!

**Comment by RobLoach** [comment]
> do you foresee any significant downsides to @madeofpeople's suggestion?

@pirog The less dependencies the better :+1: .

**Comment by pirog** [comment]
We are of the same mind then... A hive mind.

On Monday, October 6, 2014, Rob Loach notifications@github.com wrote:

> do you foresee any significant downsides to @madeofpeople
> https://github.com/madeofpeople's suggestion?
> 
> @pirog https://github.com/pirog The less dependencies the better [image:
> :+1:] .
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/kalamuna/kalastack/pull/98#issuecomment-58074537.

## 

Cheers,

Mike Pirog
Kalamuna
www.kalamuna.com

**Comment by andrewmallis** [comment]
I am running into some problems running grunt on GBZ.
Task "jasmine:themeJs" fails
Warning: PhantomJS exited unexpectedly with exit code 2.

Here is the full trace:

```
Running "jasmine:themeJs" (jasmine) task
Testing jasmine specs via PhantomJS


Running PhantomJS...ERROR
>> /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 1: /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 4w��h: not found 0 [ '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 1: /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 4w\u0001��h: not found',
>>   '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 1: /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: ����\u0007\u0003\u0002\u0002�\u0001\u00018__TEXT: not found',
>>   '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 2: /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: E�bss��\u000e��j\u0003���-\u0005��m�\u001fb\r�n��y�\u0001T: not found',
>>   '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 6: /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: Syntax error: ")" unexpected' ]
>> /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 1: /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: �����8__TEXT: not found 1 [ '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 1: /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 4w\u0001��h: not found',
>>   '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 1: /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: ����\u0007\u0003\u0002\u0002�\u0001\u00018__TEXT: not found',
>>   '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 2: /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: E�bss��\u000e��j\u0003���-\u0005��m�\u001fb\r�n��y�\u0001T: not found',
>>   '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 6: /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: Syntax error: ")" unexpected' ]
>> /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 2: /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/no�n��y�T: not found 2 [ '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 1: /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 4w\u0001��h: not found',
>>   '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 1: /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: ����\u0007\u0003\u0002\u0002�\u0001\u00018__TEXT: not found',
>>   '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 2: /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: E�bss��\u000e��j\u0003���-\u0005��m�\u001fb\r�n��y�\u0001T: not found',
>>   '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 6: /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: Syntax error: ")" unexpected' ]
>> /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 6: /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: Syntax error: ")" unexpected 3 [ '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 1: /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 4w\u0001��h: not found',
>>   '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 1: /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: ����\u0007\u0003\u0002\u0002�\u0001\u00018__TEXT: not found',
>>   '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 2: /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: E�bss��\u000e��j\u0003���-\u0005��m�\u001fb\r�n��y�\u0001T: not found',
>>   '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: 6: /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs: Syntax error: ")" unexpected' ]
Warning: PhantomJS exited unexpectedly with exit code 2. Used --force, continuing.
>> 2
>> 0 failures
Warning: Task "jasmine:themeJs" failed. Used --force, continuing.
```

I can --force it and the assets build with no diff. That's awesome.
Likely just a small tweak here?

**Comment by thiagodemellobueno** [comment]
This looks like a GBZ issue and not a kalastack issue.

What are you doing differently?
What branch are you on?
What command are you running?

I’d say this sint the right place to track this.
The jasmine tests are GBZ theme stuff, not kalastack frontend toolkit.

**Comment by pirog** [comment]
+1

On Monday, October 6, 2014, Thiago de Mello Bueno notifications@github.com
wrote:

> This looks like a GBZ issue and not a kalastack issue.
> 
> What are you doing differently?
> What branch are you on?
> What command are you running?
> 
> I’d say this sint the right place to track this.
> The jasmine tests are GBZ theme stuff, not kalastack frontend toolkit.
> 
> On Oct 6, 2014, at 3:54 PM, Andrew_Mallis <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:
> 
> > I am running into some problems running grunt on GBZ.
> > Task "jasmine:themeJs" fails
> > Warning: PhantomJS exited unexpectedly with exit code 2.
> > 
> > Here is the full trace:
> > 
> > Running "jasmine:themeJs" (jasmine) task
> > Testing jasmine specs via PhantomJS
> > 
> > Running PhantomJS...ERROR
> > 
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 1:
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 4w��h: not found 0 [
> > > > '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 1:
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 4w\u0001��h: not found',
> > > > 
> > > > '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 1:
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > ����\u0007\u0003\u0002\u0002�\u0001\u00018__TEXT: not found',
> > > > 
> > > > '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 2:
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > E�bss��\u000e��j\u0003���-\u0005��m�\u001fb\r�n��y�\u0001T: not found',
> > > > 
> > > > '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 6:
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > Syntax error: ")" unexpected' ]
> > > > 
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 1:
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > �����8__TEXT: not found 1 [
> > > > '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 1:
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 4w\u0001��h: not found',
> > > > 
> > > > '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 1:
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > ����\u0007\u0003\u0002\u0002�\u0001\u00018__TEXT: not found',
> > > > 
> > > > '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 2:
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > E�bss��\u000e��j\u0003���-\u0005��m�\u001fb\r�n��y�\u0001T: not found',
> > > > 
> > > > '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 6:
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > Syntax error: ")" unexpected' ]
> > > > 
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 2:
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/no�n��y�T:
> > > > not found 2 [
> > > > '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 1:
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 4w\u0001��h: not found',
> > > > 
> > > > '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 1:
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > ����\u0007\u0003\u0002\u0002�\u0001\u00018__TEXT: not found',
> > > > 
> > > > '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 2:
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > E�bss��\u000e��j\u0003���-\u0005��m�\u001fb\r�n��y�\u0001T: not found',
> > > > 
> > > > '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 6:
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > Syntax error: ")" unexpected' ]
> > > > 
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 6:
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > Syntax error: ")" unexpected 3 [
> > > > '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 1:
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 4w\u0001��h: not found',
> > > > 
> > > > '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 1:
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > ����\u0007\u0003\u0002\u0002�\u0001\u00018__TEXT: not found',
> > > > 
> > > > '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 2:
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > E�bss��\u000e��j\u0003���-\u0005��m�\u001fb\r�n��y�\u0001T: not found',
> > > > 
> > > > '/var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > 6:
> > > > /var/www/greenbiz/sites/all/themes/greenbiz/node_modules/grunt-contrib-jasmine/node_modules/grunt-lib-phantomjs/node_modules/phantomjs/lib/phantom/bin/phantomjs:
> > > > Syntax error: ")" unexpected' ]
> > > > Warning: PhantomJS exited unexpectedly with exit code 2. Used --force,
> > > > continuing.
> > > > 2
> > > > 0 failures
> > > > Warning: Task "jasmine:themeJs" failed. Used --force, continuing.
> > > > I can --force it and the assets build with no diff. That's awesome.
> > > > Likely just a small tweak here?
> > 
> > —
> > Reply to this email directly or view it on GitHub.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/kalamuna/kalastack/pull/98#issuecomment-58085183.

## 

Cheers,

Mike Pirog
Kalamuna
www.kalamuna.com

**Comment by andrewmallis** [comment]
I am reporting back as per instructions

```
to make sure it is consistent and integrates with our current workflows.
```

I am running grunt from within kalastack on the sprint branch.
This same command passes without warning on my local for comparison.

I have run bower install & npm install within the virtualmachine/

Yes, this is GBZ specific at this point.
I am not super aware of the JS linting+testing technologies, or if they are anticipated to be used in the base front-end stack.

**Comment by pirog** [comment]
thanks for providing the great feedback here @andrewmallis.

that said i should have been more clear in my instructions. we really only want to do a cursory overview here since there will undoubtedly be specific tasks that give us more trouble initially and on a project to project basis.

**Comment by RobLoach** [comment]
> I am not super aware of the JS linting+testing technologies, or if they are anticipated to be used in the base front-end stack.

As long as `npm` is available, then the project can define its own code syntax linting/testing using `npm test`. [ESLint](http://eslint.org/) or [JSHint](http://jshint.com/) are the popular ones.

@pirog Anything keeping this from being merged? My initial tests were fine, I'll run another one on a different machine.

**Comment by pirog** [comment]
@RobLoach re: JS linting +1... this should be per project so not in the stack

i am fine merging this in for now but @madeofpeople had some valid concerns... do we want to try to address them in this PR or maybe merge this and open a new issue?

File: modules/frontend/manifests/install.pp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/**
+ *
+ * Installs drush
```
Comment (RobLoach) [inline modules/frontend/manifests/install.pp:3]: Update docs to state that it's installing Node and Ruby?

**Comment by RobLoach** [comment]
> i am fine merging this in for now but @madeofpeople had some valid concerns... do we want to try to address them in this PR or maybe merge this and open a new issue?

After another test....

```
vagrant up
vagrant ssh
node
The program 'node' is not installed
```

Did I do something wrong?

**Comment by pirog** [comment]
@RobLoach did you follow the directions posted above and also now here:
https://github.com/kalamuna/kalastack/wiki/Adding-front-end-tooling

**Comment by pirog** [comment]
re: updating docs...

what would Will Riker do?

image

ohhh.... in that case

image

**Comment by RobLoach** [comment]
Works well! Thanks @pirog. The docs helped a lot.

**Comment by pirog** [comment]
That's why I wrote them :):)

On Sunday, October 12, 2014, Rob Loach notifications@github.com wrote:

> Works well! Thanks @pirog https://github.com/pirog. The docs helped a
> lot.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/kalamuna/kalastack/pull/98#issuecomment-58839074.

## 

Cheers,

Mike Pirog
Kalamuna
www.kalamuna.com

**Comment by pirog** [comment]
@RobLoach @andrewmallis @madeofpeople TO MERGE OR NOT TO MERGE?

only you can decide!

File: modules/frontend/manifests/install.pp (hunk 1 above)
Comment (RobLoach) [inline modules/frontend/manifests/install.pp:3]: @pirog This is the only note I have, and it surrounds code comments. Doesn't hit architecture or anything that hurts functionality. Other than that, I'd say :+1: .

**Comment by RobLoach** [comment]
@pirog 

> TO MERGE OR NOT TO MERGE?

The only comment I have is surrounding documentation: https://github.com/kalamuna/kalastack/pull/98#discussion-diff-18666937 . Other than that, looks good!

**Comment by pirog** [comment]
ok. im going to merge this and we can clean up docs in other PR!

F IT. DO IT LIVE!

**Pull request merged**